packages/nextjs/src/utils/supabaseServerClient.ts

```diff
@@ -38,6 +38,7 @@ export default function supabaseServerClient(
   }
   const access_token =
     context.req.cookies[`${cookieOptions.name}-access-token`];
```

Comment:
```suggestion
  const { value: access_token } =
    context.req.cookies.getWithOptions(`${cookieOptions.name}-access-token`);
```

Reply:
According to https://nextjs.org/docs/messages/middleware-upgrade-guide#cookies-api-revamped